tensorboard/uploader/uploader.py

```diff
@@ -157,6 +157,23 @@ def __init__(
         self._logdir_loader = logdir_loader.LogdirLoader(
             self._logdir, directory_loader_factory
         )
+        self._tracker = upload_tracker.UploadTracker(
+            verbosity=self._verbosity, one_shot=self._one_shot
+        )
+
+    @property
+    def tracker(self) -> upload_tracker.UploadTracker:
```

Comment:
To minimize exposure of implementation details, consider just making this a has_data() property and returning self._tracker.has_data().

Reply:
Done.